### PR TITLE
feat(ingestion): schema, table filtering for redshift-usage

### PR DESCRIPTION
**Pull request by Abhiram98**
Filter redshift usage statistics based on schema and table patterns.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by Abhiram98** [comment]
Trying to implement this feature request:
https://feature-requests.datahubproject.io/b/feedback/p/redshift-usage-support-table-and-schema-filtering

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py [hunk 1]
```diff
@@ -223,7 +223,12 @@ def _get_redshift_history(
                 event_dict["endtime"] = event_dict.get("endtime").__str__()
 
             logger.debug(f"event_dict: {event_dict}")
-            events.append(event_dict)
+            # filter based on schema and table pattern
+            if self.config.schema_pattern.allowed(event_dict['schema'])\
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:227]: any plans to include `view_pattern` as well?

File: metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py [hunk 2]
```diff
@@ -82,6 +82,54 @@ def test_redshift_usage_source(pytestconfig, tmp_path):
...
+                    "type": "redshift-usage",
+                    "config": {
+                        "host_port": "xxxxx",
+                        "database": "xxxxx",
+                        "username": "xxxxx",
+                        "password": "xxxxx",
+                        "email_domain": "acryl.io",
+                        "include_views": True,
+                        "include_tables": True,
+                        "schema_pattern": {
```
Comment (rslanka) [inline metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py:112]: why not include a `table_pattern` as well to this test?

**Comment by rslanka** [comment]
@Abhiram98, looks great overall! It would be great if you could address the comments above.

File: metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py (hunk 2 above)
Comment (Abhiram98) [reply metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py:112]: Good point, I will include this as well

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py (hunk 1 above)
Comment (Abhiram98) [reply metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:227]: @rslanka, it seems like the script wasn't ingesting view related usage stats in the first place. I've looked at the code and couldn't find anything. Did a small test run as well; view usage didn't get ingested.

File: metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py (hunk 2 above)
Comment (Abhiram98) [reply metadata-ingestion/tests/integration/redshift-usage/test_redshift_usage.py:112]: Added this test case

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py [hunk 3]
```diff
@@ -223,7 +223,18 @@ def _get_redshift_history(
...
+            # filter based on schema and table pattern
+            if (
+                event_dict["schema"]
+                and self.config.schema_pattern.allowed(event_dict["schema"])
+                and event_dict["table"]
+                and self.config.table_pattern.allowed(event_dict["table"])
+            ):
+                events.append(event_dict)
+            else:
+                logger.debug(
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:235]: Please add this to the source report as well. See [this example] (https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py#L187)

**Review by rslanka — CHANGES_REQUESTED** [review]
Looks good. @Abhiram98 , could you please address the change requested [here](https://github.com/datahub-project/datahub/pull/4396/files#r833548677) (and we can merge)?

File: metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py (hunk 3 above)
Comment (Abhiram98) [reply metadata-ingestion/src/datahub/ingestion/source/usage/redshift_usage.py:245]: Hey @rslanka, I have added this to the reporting as well

**Review by rslanka — APPROVED** [review]
LGTM!

**Pull request merged**